### PR TITLE
helm chart: add nginx to avoid CORS issue

### PR DESCRIPTION
**Pull request by Ruoyu-y**
## Description

To avoid CORS issue, add nginx within the helm chart based pipeline.

## Issues

#368 

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

`n/a`

## Tests

`n/a`

### CODE REVIEW

File: helm-charts/chatqna/templates/nginx-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        location /v1/chatqna {
+            proxy_pass http://{{ include "chatqna.fullname" . }}:{{ .Values.service.port }};
+            proxy_set_header Host $host;
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto $scheme;
+        }
+
+        location /v1/dataprep {
+            proxy_pass http://{{ include "data-prep.serviceName" . }}:{{ index .Values "data-prep" "service" "port" }};
```
[the remark this answers is not in the document]
Comment (Ruoyu-y) [reply helm-charts/chatqna/templates/nginx-deployment.yaml:33]: Due to that the subchart name contains '-', simply using '.Subcharts.data-prep' will return "bad character U+002D '-'" error. So still put the definition within _helpers.tpl file

File: helm-charts/chatqna/values.yaml [hunk 2]
```diff
@@ -39,6 +39,17 @@ affinity: {}
...
 
+# To override values in subchart chatqna-ui
+chatqna-ui:
+  # switch to traditional UI image if needed
+  image:
+    repository: "opea/chatqna-conversation-ui"
+    tag: "latest"
+  # switch containerPort to 5173 if using traditional UI
+  containerPort: "80"
+  BACKEND_SERVICE_ENDPOINT: http://localhost:30123/v1/chatqna
```
[the remark this answers is not in the document]
Comment (Ruoyu-y) [reply helm-charts/chatqna/values.yaml]: > should be mega serivce URL and port?

As mega service and data prep service are still internal services behind nginx, put nginx ip and port here instead.

**Review by yongfengdu — APPROVED** [review]

**Comment by lianhao** [comment]
Why this doesn't trigger helm CI test?

**Comment by lianhao** [comment]
@Ruoyu-y  please fix the helm chatqna-ui test failure

**Review by lianhao — APPROVED** [review]

**Pull request merged**